cdk/lib/nishikiBackend-stack.ts

```diff
@@ -179,5 +179,14 @@ const nishikiMainBackendFunction = (
 		},
 	);
 
+	mainFunction.addToRolePolicy(
+		new iam.PolicyStatement({
+			actions: ["events:PutEvents"],
+			resources: [
+				`arn:aws:events:${this.region}:${this.account}:event-bus/NishikiNotificationBus`,
```

Comment:
This line causes an error when it's compiled.
```
TSError: ⨯ Unable to compile TypeScript:
lib/nishikiBackend-stack.ts:186:23 - error TS2532: Object is possibly 'undefined'.

186     `arn:aws:events:${this.region}:${this.account}:event-bus/NishikiNotificationBus`,
```

So instead of `this`, I think scope should be here.
```typescript
const nishikiMainBackendFunction = (
	scope: cdk.Stack,
```

```suggestion
				`arn:aws:events:${scope.region}:${scope.account}:event-bus/NishikiNotificationBus`,
```